### PR TITLE
update vscode typings

### PR DESCRIPTION
**Pull request by baronfel**
pretty straightforward: update vscode dependency and update the bindings to it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   },
   "homepage": "https://github.com/ionide/ionide-vscode-helpers#readme",
   "scripts": {
+    "gen:vscode": "ts2fable node_modules/@types/vscode/index.d.ts src/Fable.Import.VSCode.fs"
```
Comment (Krzysztof-Cieslak) [inline package.json:28]: maybe we could add something like `"gen:vscode-proposed": "vscode-dts dev && ts2fable (whatever-is-the-name-of-the-file-i-dont-remember.d.ts) src/Fable.Import.VSCode.Proposed.fs`

Comment (baronfel) [reply package.json:28]: certainly! I'm currently doing a WIP conversion of ionide to use these new generated helpers to make sure things don't puke too badly, so once I've done that to validate this core idea I'd be glad to add more

Comment (baronfel) [reply package.json:28]: sidenote: Thenables are a pain in the ass to work with in Fable.  I need a way to lift them to the Promise<'t> shape.

**Comment by baronfel** [comment]
I'm happy with how these look and work, so I'm going to merge this.

**Comment by Krzysztof-Cieslak** [comment]
🎉

**Pull request merged**